### PR TITLE
wander: 0.7.0 -> 0.8.0

### PR DESCRIPTION
**Pull request by dbalan**
###### Description of changes

Wander new release, changelogs https://github.com/robinovitch61/wander/releases/tag/v0.8.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by superherointj** [comment]
Commit message should be: `wander: 0.7.0 -> 0.8.0`

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 189102` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wander</li>
  </ul>
</details>

**Comment by superherointj** [comment]
* Shell completion is missing.
* Version test is missing.

**Review by astro — APPROVED** [review]

**Comment by superherointj** [comment]
@astro The application might run. But package is still not right.

**Comment by superherointj** [comment]
@dbalan ping

**Review by astro — CHANGES_REQUESTED** [review]
what @superherointj said

**Comment by dbalan** [comment]
@astro @superherointj Sorry for the late response, I have updated the PR with the changes :)

File: pkgs/tools/admin/wander/default.nix [hunk 1]
```diff
@@ -1,17 +1,33 @@
-{ buildGoModule, fetchFromGitHub, lib }:
+{ buildGoModule, fetchFromGitHub, installShellFiles, lib }:
 
 buildGoModule rec {
   pname = "wander";
-  version = "0.7.0";
+  version = "0.8.0";
+
+  # wander embeds commitSHA into version information
+  commitSHA = "94e61b57f353b7b28abf9c9d088319a2d98ecbfa";
```
Comment (superherointj) [inline pkgs/tools/admin/wander/default.nix:8]:
```suggestion
```
The bot won't be able to update this commitSHA unless if you make an update script. But then, you'd increase the upkeep burden for maintainers (update scripts are messy). I suggest you leave it out.

File: pkgs/tools/admin/wander/default.nix [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
+    sha256 = "sha256-G/TrfnmEyomdUCN5nUS9v5iqeUzgZzMLUZnfroQLZuk=";
   };
 
-  vendorSha256 = "sha256-T+URnRLumXFz48go9TN0Wha99T03OWGfDK7cQ+zKeRI=";
+  vendorSha256 = "sha256-iTaZ5/0UrLJ3JE3FwQpvjKKrhqklG4n1WFTJhWfj/rI=";
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  ldflags = [ "-X github.com/robinovitch61/wander/cmd.Version=v${version}"
+              "-X github.com/robinovitch61/wander/cmd.CommitSHA=${commitSHA}" ];
```
Comment (superherointj) [inline pkgs/tools/admin/wander/default.nix:23]:
```suggestion
  ldflags = [ "-X github.com/robinovitch61/wander/cmd.Version=v${version}" ];
```

Comment (superherointj) [reply pkgs/tools/admin/wander/default.nix:23]: Also feel free (optional) to add a version test. To future proof for changes in upstream.

Comment (superherointj) [reply pkgs/tools/admin/wander/default.nix:23]: Notice, `ldflags` is configure phase. `nativeBuildInputs` is buildPhase.
configure phase comes before build phase.

File: pkgs/tools/admin/wander/default.nix [hunk 3]
```diff
@@ -1,17 +1,33 @@
...
+
+  postInstall = ''
+    installShellCompletion --cmd wander \
+                           --fish <($out/bin/wander completion fish) \
+                           --bash <($out/bin/wander completion bash) \
+                           --zsh <($out/bin/wander completion zsh)
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/wander/default.nix:29]:
```suggestion
    installShellCompletion --cmd wander \
      --fish <($out/bin/wander completion fish) \
      --bash <($out/bin/wander completion bash) \
      --zsh <($out/bin/wander completion zsh)
```

File: pkgs/tools/admin/wander/default.nix (hunk 1 above)
Comment (dbalan) [reply pkgs/tools/admin/wander/default.nix]: I have removed it, but had to patch `wander` (unset commitSHA would panic at startup)

**Comment by dbalan** [comment]
Updated with requested changes @SuperSandro2000 @superherointj

File: pkgs/tools/admin/wander/default.nix [hunk 4]
```diff
@@ -1,17 +1,40 @@
...
+  postInstall = ''
+    installShellCompletion --cmd wander \
+      --fish <($out/bin/wander completion fish) \
+      --bash <($out/bin/wander completion bash) \
+      --zsh <($out/bin/wander completion zsh)
+  '';
+
+  passthru.tests.version = testers.testVersion {
+    package = wander;
+    command = "wander --version";
```
Comment (superherointj) [inline pkgs/tools/admin/wander/default.nix:36]: To test version, run:
> nix build .#wander.passthru.tests.version

It is failing...

```suggestion
    command = "wander --version";
    version = "v${version}";
```

**Review by superherointj — APPROVED** [review]

**Pull request merged**